src/app/pages/request-query/request-query.component.ts

```diff
@@ -91,14 +101,24 @@ export class RequestQueryComponent implements OnInit, IModelPage {
           ]);
         }),
         catchError(() => {
-          this.toastr.error('There was a problem finding your request queries.');
-          this.status = 'error';
-          return EMPTY;
+          return this.errorResponse('There was a problem finding your request queries.');
+        }),
+        switchMap(([dataElements, query]) => {
+          return this.queryBuilderService.setupConfig(
+            this.mapDataElements(dataElements),
+            query
+          );
+        }),
+        catchError(() => {
+          return this.errorResponse(
+            'There was a problem configuring your request queries.'
+          );
         })
       )
-      .subscribe(([dataElements, query]) => {
-        this.dataElements = this.mapDataElements(dataElements);
-        this.query = query;
+      .subscribe(([queryConfiguration]) => {
```

Comment:
Linked to my comment about the return type - if just the `Observable<QueryConfiguration>` type was returned, then this destructuring wouldn't be necessary.

Reply:
Updated as suggested.